### PR TITLE
REL-3176: Port giapi-glue-cc to run on CentOS 7

### PR DESCRIPTION
**Pull request by cquiroz**
This PR updates the giapi-glue-cc to make it run on CentOS and it fixes a few warnings produced by the compiler

### CODE REVIEW

File: conf/common.mk [hunk 1]
```diff
@@ -74,6 +74,6 @@ TMP_DIST_DIR := /tmp/$(DIST_PACKAGE_NAME)
 %.o: %.cpp
 	@echo 'Building file: $<'
 	@echo 'Invoking $(OS) C++ Compiler'
-	$(CXX) $(INC_DIRS) -g -O0 -Wall -fPIC -c -fmessage-length=0 -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o"$@" "$<"
+	$(CXX) $(INC_DIRS) -g -O0 -Wall -fPIC -c -Wno-deprecated -fmessage-length=0 -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o"$@" "$<"
```
Comment (swalker2m) [inline conf/common.mk:77]: `-Wno-deprecated` ?

File: src/status/StatusItem.cpp [hunk 2]
```diff
@@ -27,6 +27,12 @@ StatusItem::StatusItem(const std::string &name, const type::Type type) :
 	case type::INT:
 		_value = 0;
 		break;
+	case type::BYTE:
+		_value = (unsigned char)0;
+		break;
+	case type::SHORT:
+		_value = (unsigned short)0;
```
Comment (swalker2m) [inline src/status/StatusItem.cpp:34]: What's going on here with assigning `_value` all these different types?

**Review by swalker2m — APPROVED** [review]
The last time I worked with C++ and Makefiles was probably in the 90s.  👍

File: conf/common.mk (hunk 1 above)
Comment (cquiroz) [reply conf/common.mk:77]: This removes a warning of using a deprecated module. The correct way would be to remove the usage of those deprecated methods but I consider that out of scope

File: src/status/StatusItem.cpp (hunk 2 above)
Comment (cquiroz) [reply src/status/StatusItem.cpp:34]: Yeah, that's a bit odd. this are status items and they can be different types like EPICS channels so we have a field to store the type and the actual value as a `boost` `any`. Every time we reach for the value we must check the type

BYTE and SHORT were not considered so it was giving a warning about non exhaustive `switch`

**Pull request merged**